pkg/cmd/kind/load/docker-image/docker-image.go

```diff
@@ -141,6 +141,11 @@ func runE(logger log.Logger, flags *flagpole, args []string) error {
 		}
 	}
 
+	// return early if no node needs the image
```

Comment:
unrelated to this PR but the code above also looks suspect ... where we check `len(selectedNodes)` when logging about a particular image even though nodes may have been selected for another image, will address in a follow-up.

Reply:
selectedNodes also really ought to be a set, clearly should have reviewed this code more carefully.